src/h5web/dataset-visualizer/DimensionMapper.tsx

```diff
@@ -83,9 +87,23 @@ function DimensionMapper(props: Props): JSX.Element {
 
   return (
     <div className={styles.mapper}>
-      <div className={styles.axisMapperWrapper}>{renderAxisMapper('x')}</div>
-      <div className={styles.sliderWrapper}>
-        {mapperState.map(renderSlicingSlider)}
+      <div className={styles.axisMapperWrapper}>
+        <div className={styles.dims}>
+          <span className={styles.dimsLabel}>n</span>
+          {rawDims.map((d, i) => (
+            // eslint-disable-next-line react/no-array-index-key
+            <span key={`${i}${d}`} className={styles.dimSize}>
+              {d}
+            </span>
+          ))}
+        </div>
+        {renderAxisMapper('x')}
+        {mapperState.includes('y') && renderAxisMapper('y')}
+      </div>
+      <div className={styles.sliders}>
+        {mapperState.map((slicingIndex, dimIndex) =>
+          renderSlicingSlider(slicingIndex, dimIndex)
+        )}
```

Comment:
You can simplify this to `mapperState.map(renderSlicingSlider)`

Reply:
I wish but TS complains because of the `key` optional argument of `renderSlicingSlider` :shrug:

Reply:
I think that's because of the extra `key` argument.